validator_worker/src/sentry_interface.rs

```diff
@@ -225,23 +227,19 @@ fn handle_http_error(e: reqwest::Error, url: &str) {
     }
 }
 
-pub async fn all_channels(
-    sentry_url: &str,
-    adapter: impl Adapter + 'static,
-) -> Result<Vec<Channel>, ()> {
-    let validator = adapter.whoami();
+pub async fn all_channels(sentry_url: &str, whoami: String) -> Result<Vec<Channel>, ()> {
     let url = sentry_url.to_owned();
-    let first_page = fetch_page(url.clone(), 0, validator.clone())
+    let first_page = fetch_page(url.clone(), 0, whoami.clone())
         .await
         .expect("Failed to get channels from sentry url");
     if first_page.total_pages < 2 {
         Ok(first_page.channels)
     } else {
         let mut all: Vec<ChannelAllResponse> = try_join_all(
-            (0..first_page.total_pages).map(|i| fetch_page(url.clone(), i, validator.clone())),
+            (0..first_page.total_pages).map(|i| fetch_page(url.clone(), i, whoami.clone())),
         )
         .await
-        .unwrap();
+        .expect("Failed to fetch all channels");
```

Comment:
Shouldn't this be returned as an error, instead of crashing the validator? Again - Sentry down for a bit, network issues and etc., will cause it to blow up and stop the validator.

Reply:
This could be helpful in cases of wrong sentry url